### PR TITLE
feat(TCOMP-2408): all causes of ComponentException are Exception.class

### PR DESCRIPTION
**Pull request by ypiel-talend**
### Requirements

* Any code change adding any logic **MUST** be tested through a unit test executed with the default build
* Any API addition **MUST** be done with a documentation update if relevant 

### Why this PR is needed?
https://jira.talendforge.org/browse/TCOMP-2408

### What does this PR adds (design/code thoughts)?
All cause exceptions are transformed to Exception.class to be known by all runtimes.

### CODE REVIEW

File: component-api/src/main/java/org/talend/sdk/component/api/exception/ComponentException.java [hunk 1]
```diff
@@ -70,4 +71,23 @@ public ComponentException(final Throwable cause) {
         this(ErrorOrigin.UNKNOWN, cause.getMessage(), cause);
     }
 
+    /**
+     * Transform all specific cause exceptions to Exception class.
+     * @param t
+     * @return An exception of type java.lang.Exception.
+     */
+    private Exception transformException(Throwable t) {
```
Comment (undx) [inline component-api/src/main/java/org/talend/sdk/component/api/exception/ComponentException.java:79]: final modifier.

**Review by undx — DISMISSED** [review]

File: sample-parent/tck-validator/pom.xml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    <dependency>
+      <groupId>org.talend.sdk.component</groupId>
+      <artifactId>component-api</artifactId>
+      <version>1.65.0M4-SNAPSHOT</version>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.talend.sdk.component</groupId>
+      <artifactId>component-runtime-impl</artifactId>
+      <version>1.65.0M4-SNAPSHOT</version>
```
Comment (undx) [inline sample-parent/tck-validator/pom.xml:44]: use `${project.version}`

File: sample-parent/tck-validator/README.adoc [hunk 3]
```diff
@@ -0,0 +1,2 @@
+= A tck sample Family
```
Comment (undx) [inline sample-parent/tck-validator/README.adoc:1]: purpose?

File: component-api/src/main/java/org/talend/sdk/component/api/exception/ComponentException.java [hunk 4]
```diff
@@ -70,4 +71,35 @@ public ComponentException(final Throwable cause) {
         this(ErrorOrigin.UNKNOWN, cause.getMessage(), cause);
     }
 
+    /**
+     * Convert all exception stack to generic throwable stack to avoid unknown exception at deserialization time..
+     * 
+     * @param t
+     * @return An Throwable.
+     */
+    protected Throwable toGenericThrowable(final Throwable t) {
```
Comment (undx) [inline component-api/src/main/java/org/talend/sdk/component/api/exception/ComponentException.java:80]: this method looks like InvocationExceptionWrapper.mapException, no? ;-)

File: component-api/src/test/java/org/talend/sdk/component/api/exception/ComponentExceptionTest.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    void testParameters() {
+        ComponentException componentException = new ComponentException(new IndexOutOfBoundsException("cause"));
+        Assertions.assertNotNull(componentException.getCause());
+        Assertions.assertEquals("java.lang.IndexOutOfBoundsException", componentException.getOriginalType());
+        Assertions.assertEquals(ComponentException.ErrorOrigin.UNKNOWN, componentException.getErrorOrigin());
+
+        componentException = new ComponentException(ComponentException.ErrorOrigin.USER, "message");
+        Assertions.assertEquals(ComponentException.ErrorOrigin.USER, componentException.getErrorOrigin());
+    }
+
```
Comment (undx) [inline component-api/src/test/java/org/talend/sdk/component/api/exception/ComponentExceptionTest.java:60]: You could also inspire you from InvocationExceptionWrapperTest class...

**Review by undx — COMMENTED** [review]
Some comments. 
Also, I don't understand the new connector added to module sample-parent purpose...
Could you explain why adding another connector in the context of this pull-request?
(seen the GenerateExeceptionService but don't see the usage)

File: sample-parent/tck-validator/README.adoc (hunk 3 above)
Comment (yyin-talend) [reply sample-parent/tck-validator/README.adoc:1]: Used in the Test case, discussed with Axel, or, only used in test case, not commit?

File: component-api/src/main/java/org/talend/sdk/component/api/exception/ComponentException.java (hunk 4 above)
Comment (yyin-talend) [reply component-api/src/main/java/org/talend/sdk/component/api/exception/ComponentException.java:80]: InvocationExceptionWrapper.mapException mainly for : toRuntimeException, not similar

Comment (yyin-talend) [reply component-api/src/main/java/org/talend/sdk/component/api/exception/ComponentException.java:80]: InvocationExceptionWrapper.mapException mainly for : toRuntimeException, not similar

File: component-api/src/test/java/org/talend/sdk/component/api/exception/ComponentExceptionTest.java (hunk 5 above)
Comment (yyin-talend) [reply component-api/src/test/java/org/talend/sdk/component/api/exception/ComponentExceptionTest.java:60]: Line 62-66 is hard to execute into...

**Review by undx — COMMENTED** [review]
As said in daily, please integrate relevant changes in existing sample connector.

**Review by acatoire — CHANGES_REQUESTED** [review]
You have 2 sonar Code Smells detected; 
Are they relevant?

**Review by undx — APPROVED** [review]
lgtm

**Review by acatoire — APPROVED** [review]

**Pull request merged**